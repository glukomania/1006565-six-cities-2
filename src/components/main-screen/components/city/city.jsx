import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {ActionCreator} from '../../../../store/reducer';

const City = (props) => {
  const {city, onCityClick, currentCity} = props;

  const cityClickHandler = () => {
    onCityClick(city);
    // const target = evt.target;
    // target.parentNode.classList.add(`tabs__item--active`);
  };

  return <li className="locations__item" onClick={cityClickHandler}>
    <a className={city === currentCity ? `locations__item-link tabs__item tabs__item tabs__item--active` : `locations__item-link`} href="#">
      <span>{city}</span>
    </a>
  </li>;
};

City.propTypes = {
  city: PropTypes.string.isRequired,
  onCityClick: PropTypes.func,
  currentCity: PropTypes.string.isRequired
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  currentCity: state.currentCity,
});

const mapDispatchToProps = (dispatch) => ({
  onCityClick: (value) => {
    dispatch(ActionCreator.changeCity(value));
    dispatch(ActionCreator.changeCoords(value));
    dispatch(ActionCreator.getOffers(value));
  }
});

export {City};
export default connect(mapStateToProps, mapDispatchToProps)(City);
