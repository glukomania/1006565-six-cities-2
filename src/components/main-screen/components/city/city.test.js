import renderer from 'react-test-renderer';
import {City} from './city';

it(`City is displayed correctly`, () => {
  const city = renderer
    .create(<City
      city={`Test`}
    />)
    .toJSON();

  expect(city).toMatchSnapshot();
});
