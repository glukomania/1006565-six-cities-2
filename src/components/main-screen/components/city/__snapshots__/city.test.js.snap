// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City is displayed correctly 1`] = `
<li
  className="locations__item"
  onClick={[Function]}
>
  <a
    className="locations__item-link"
    href="#"
  >
    <span>
      Test
    </span>
  </a>
</li>
`;
